subsys/bluetooth/host/classic/avrcp.c

```diff
@@ -2475,6 +2738,185 @@ int bt_avrcp_tg_send_set_browsed_player_rsp(struct bt_avrcp_tg *tg, uint8_t tid,
+		break;
+	case BT_AVRCP_EVT_VOLUME_CHANGED:
+		param_len += sizeof(data->absolute_volume);
+		break;
+	case BT_AVRCP_EVT_TRACK_REACHED_END:
+	case BT_AVRCP_EVT_TRACK_REACHED_START:
+	case BT_AVRCP_EVT_AVAILABLE_PLAYERS_CHANGED:
+	case BT_AVRCP_EVT_NOW_PLAYING_CONTENT_CHANGED:
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	if (net_buf_tailroom(buf) < param_len) {
+		LOG_ERR("Not enough space in net_buf");
+		return -ENOMEM;
+	}
+
+	net_buf_add_u8(buf, event_id);
+	switch (event_id) {
+	case BT_AVRCP_EVT_PLAYBACK_STATUS_CHANGED:
+		if (data->play_status > BT_AVRCP_PLAYBACK_STATUS_REV_SEEK &&
+		    data->play_status != BT_AVRCP_PLAYBACK_STATUS_ERROR) {
+			LOG_ERR("Invalid playback status: %d", data->play_status);
+			return -EINVAL;
+		}
+		net_buf_add_u8(buf, data->play_status);
+		break;
+	case BT_AVRCP_EVT_TRACK_CHANGED:
+		uint64_t identifier = sys_get_be64(data->identifier);
+
+		net_buf_add_be64(buf, identifier);
+		break;
+	case BT_AVRCP_EVT_PLAYBACK_POS_CHANGED:
+		net_buf_add_be32(buf, data->playback_pos);
+		break;
+	case BT_AVRCP_EVT_BATT_STATUS_CHANGED:
+		if (data->battery_status > BT_AVRCP_BATTERY_STATUS_FULL) {
+			LOG_ERR("Invalid battery status: %d", data->battery_status);
+			return -EINVAL;
+		}
+		net_buf_add_u8(buf, data->battery_status);
+		break;
+	case BT_AVRCP_EVT_SYSTEM_STATUS_CHANGED:
+		if (data->system_status > BT_AVRCP_SYSTEM_STATUS_UNPLUGGED) {
+			LOG_ERR("Invalid system status: %d", data->system_status);
+			return -EINVAL;
+		}
+		net_buf_add_u8(buf, data->system_status);
+		break;
+	case BT_AVRCP_EVT_PLAYER_APP_SETTING_CHANGED:
+		net_buf_add_u8(buf, data->setting_changed.num_of_attr);
+		net_buf_add_mem(buf, data->setting_changed.attr_vals,
+		data->setting_changed.num_of_attr * sizeof(struct bt_avrcp_app_setting_attr_val));
+		break;
+	case BT_AVRCP_EVT_ADDRESSED_PLAYER_CHANGED:
+		net_buf_add_be16(buf, data->addressed_player_changed.player_id);
+		net_buf_add_be16(buf, data->addressed_player_changed.uid_counter);
+		break;
+	case BT_AVRCP_EVT_UIDS_CHANGED:
+		net_buf_add_be16(buf, data->uid_counter);
+		break;
+	case BT_AVRCP_EVT_VOLUME_CHANGED:
+		if (data->absolute_volume > BT_AVRCP_MAX_ABSOLUTE_VOLUME) {
+			LOG_ERR("Invalid absolute volume: %d", data->absolute_volume);
+			return -EINVAL;
+		}
+		net_buf_add_u8(buf, data->absolute_volume);
+		break;
+	case BT_AVRCP_EVT_TRACK_REACHED_END:
+	case BT_AVRCP_EVT_TRACK_REACHED_START:
+	case BT_AVRCP_EVT_AVAILABLE_PLAYERS_CHANGED:
+	case BT_AVRCP_EVT_NOW_PLAYING_CONTENT_CHANGED:
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+int bt_avrcp_tg_send_notification_rsp(struct bt_avrcp_tg *tg, uint8_t tid, uint8_t status,
+				      uint8_t event_id, struct bt_avrcp_event_data *data)
+{
+	uint8_t rsp_code = BT_AVRCP_RSP_INTERIM;
+	struct net_buf *buf;
+	int err;
+
+	if ((tg == NULL) || (tg->avrcp == NULL)) {
+		return -EINVAL;
+	}
+
+	if (!IS_TG_ROLE_SUPPORTED()) {
+		return -ENOTSUP;
+	}
+
+	switch (status) {
```

Comment:
Is it needed: save the flag whether the event is registered, then return error if not registered by CT?